runtime/tests/CMakeLists.txt

```diff
@@ -26,7 +26,7 @@ add_executable(runner_tests runner_main.cpp)
 
 add_subdirectory(third_party)
 
-if (ENABLE_ADDRESS_SANITIZER)
+if (ENABLE_ADDRESS_SANITIZER OR APPLE)
```

Comment:
Is this correct?

```suggestion
# Skip dynamic device loading test when sanitizers are enabled or when building on mac, due to
# RTLD_DEEPBIND incompatibility.
if (ENABLE_ADDRESS_SANITIZER OR APPLE)
```

Reply:
I've updated your suggestion! I'll revisit the dynamic device loading work on macOS as the current support is not fully compatible with the macOS dylib API.

Reply:
Thanks!